app/src/main/java/io/homeassistant/companion/android/settings/vehicle/views/AndroidAutoFavoritesView.kt

```diff
@@ -0,0 +1,132 @@
+package io.homeassistant.companion.android.settings.vehicle.views
+
+import android.os.Build
+import androidx.annotation.RequiresApi
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import io.homeassistant.companion.android.common.data.integration.Entity
+import io.homeassistant.companion.android.common.data.integration.friendlyName
+import io.homeassistant.companion.android.database.server.Server
+import io.homeassistant.companion.android.settings.vehicle.ManageAndroidAutoViewModel
+import io.homeassistant.companion.android.util.compose.FavoriteEntityRow
+import io.homeassistant.companion.android.util.compose.ServerExposedDropdownMenu
+import io.homeassistant.companion.android.util.compose.SingleEntityPicker
+import io.homeassistant.companion.android.vehicle.MainVehicleScreen
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import org.burnoutcrew.reorderable.ReorderableItem
+import org.burnoutcrew.reorderable.rememberReorderableLazyListState
+import org.burnoutcrew.reorderable.reorderable
+import io.homeassistant.companion.android.common.R as commonR
+
+@RequiresApi(Build.VERSION_CODES.O)
+@Composable
+fun AndroidAutoFavoritesSettings(
+    androidAutoViewModel: ManageAndroidAutoViewModel,
+    serversList: List<Server>,
+    defaultServer: Int
+) {
+    val reorderState = rememberReorderableLazyListState(
+        onMove = { from, to -> androidAutoViewModel.onMove(from, to) },
+        canDragOver = { draggedOver, _ -> androidAutoViewModel.canDragOver(draggedOver) },
+        onDragEnd = { _, _ -> androidAutoViewModel.saveFavorites() }
+    )
+
+    var selectedServer by remember { mutableStateOf(defaultServer) }
+
+    val favoriteEntities = androidAutoViewModel.favoritesList.toList()
+    var validEntities by remember { mutableStateOf<List<Entity<*>>>(emptyList()) }
+    LaunchedEffect(favoriteEntities.size, androidAutoViewModel.sortedEntities.size) {
```

Comment:
This also needs the server ID in case the user has two servers with identical numbers of entities.

```suggestion
    LaunchedEffect(favoriteEntities.size, androidAutoViewModel.sortedEntities.size, selectedServer) {
```

Reply:
ah yes thats good!